### PR TITLE
Issue #0000: fix Setting the identifier from input request to node id…

### PR DESCRIPTION
**Pull request by rheafernandes**
…entifier

### CODE REVIEW

File: ontology-engine/graph-engine_2.11/src/main/scala/org/sunbird/graph/schema/validator/BaseDefinitionNode.scala [hunk 1]
```diff
@@ -44,6 +44,8 @@ class BaseDefinitionNode(graphId: String, objectType: String, version: String =
         // TODO: set SYS_NODE_TYPE, FUNC_OBJECT_TYPE
         node.setNodeType(SystemNodeTypes.DATA_NODE.name)
         node.setObjectType(objectType)
+        node.setIdentifier(input.getOrDefault("identifier", "").asInstanceOf[String])
```
Comment (maheshkumargangula) [inline ontology-engine/graph-engine_2.11/src/main/scala/org/sunbird/graph/schema/validator/BaseDefinitionNode.scala]:
```suggestion
        node.setIdentifier(input.getOrDefault("identifier", Identifier.getIdentifier(graphId, Identifier.getUniqueIdFromTimestamp)).asInstanceOf[String])
```

File: ontology-engine/graph-engine_2.11/src/main/scala/org/sunbird/graph/schema/validator/BaseDefinitionNode.scala [hunk 2]
```diff
@@ -44,6 +44,8 @@ class BaseDefinitionNode(graphId: String, objectType: String, version: String =
...
+        input.remove("identifier")
         if (StringUtils.isBlank(node.getIdentifier)) node.setIdentifier(Identifier.getIdentifier(graphId, Identifier.getUniqueIdFromTimestamp))
```
Comment (maheshkumargangula) [inline ontology-engine/graph-engine_2.11/src/main/scala/org/sunbird/graph/schema/validator/BaseDefinitionNode.scala]:
```suggestion
```

File: ontology-engine/graph-engine_2.11/src/main/scala/org/sunbird/graph/schema/validator/BaseDefinitionNode.scala (hunk 1 above)
Comment (rheafernandes) [reply ontology-engine/graph-engine_2.11/src/main/scala/org/sunbird/graph/schema/validator/BaseDefinitionNode.scala]: Fixed it

File: ontology-engine/graph-engine_2.11/src/main/scala/org/sunbird/graph/schema/validator/BaseDefinitionNode.scala (hunk 2 above)
Comment (rheafernandes) [reply ontology-engine/graph-engine_2.11/src/main/scala/org/sunbird/graph/schema/validator/BaseDefinitionNode.scala]: Fixed this

**Pull request merged**